`views/document.template.mjs` isn't immediately available

nitro _can_ throw a mix of errors on first build (when running `nu dev`). This eventually resolves but it spams the console and is due to the template not being in vfs

```
import htmlTemplate from '/framework/playground/.nuxt/views/document.template.mjs';
       ^^^^^^^^^^^^
SyntaxError: The requested module '/framework/playground/.nuxt/views/document.template.mjs' does not provide an export named 'default'
    at ModuleJob._instantiate (internal/modules/esm/module_job.js:120:21)
    at async ModuleJob.run (internal/modules/esm/module_job.js:165:5)
    at async Loader.import (internal/modules/esm/loader.js:177:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
```
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/framework/playground/.nuxt/views/document.template.mjs' imported from /framework/playground/.nuxt/nitro/index.mjs
    at finalizeResolution (internal/modules/esm/resolve.js:271:11)
    at moduleResolve (internal/modules/esm/resolve.js:694:10)
    at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:805:11)
    at Loader.resolve (internal/modules/esm/loader.js:88:40)
    at Loader.getModuleJob (internal/modules/esm/loader.js:241:28)
    at ModuleWrap.<anonymous> (internal/modules/esm/module_job.js:72:40)
    at link (internal/modules/esm/module_job.js:71:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

--- a/packages/nitro/src/build.ts
+++ b/packages/nitro/src/build.ts
@@ -47,7 +47,7 @@ export async function build (nitroContext: NitroContext) {
   // Compile html template
   const htmlSrc = resolve(nitroContext._nuxt.buildDir, `views/${{ 2: 'app', 3: 'document' }[2]}.template.html`)
   const htmlTemplate = { src: htmlSrc, contents: '', dst: '' }
-  htmlTemplate.dst = htmlTemplate.src.replace(/.html$/, '.mjs').replace('app.', 'document.')
+  htmlTemplate.dst = htmlTemplate.src.replace(/.html$/, '.mjs').replace('app.template.mjs', 'document.template.mjs')
   htmlTemplate.contents = nitroContext.vfs[htmlTemplate.src] || await fse.readFile(htmlTemplate.src, 'utf-8')
   await nitroContext._internal.hooks.callHook('nitro:document', htmlTemplate)
   const compiled = 'export default ' + serializeTemplate(htmlTemplate.contents)

